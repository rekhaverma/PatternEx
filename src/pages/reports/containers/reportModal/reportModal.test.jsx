import React from 'react';
import ReportModal from './reportModal.jsx';

describe('<ReportModal /> page: ', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      'onClose': jest.fn(),
      'pipelines': [],
      'saveDetail': jest.fn(),
      'data': {
        'rules': '',
      },
      'updateReportFields': jest.fn(),
      'rules': [],
      'allowedFields': {
        'name': true,
        'mode': true,
        'pipelines': true,
        'rules': true,
      },
      'title': '',
      'formErrors': {},
      'formValid': false,
      'onAddnewRule': jest.fn(),
    };
    wrapper = shallow(<ReportModal {...props} />);
  });

  describe('render: ', () => {
    it('should match with snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('action simulation: ', () => {
    it('should call "saveDetail" function on click of save', () => {
      wrapper.find('.button--success').simulate('click');
      expect(props.saveDetail.mock.calls.length).toBe(1);
    });

    it('should call "onClose" function on click of cancel and close sign', () => {
      wrapper.find('.button--dark').simulate('click');
      wrapper.find('.icon-close').simulate('click');
      expect(props.onClose.mock.calls.length).toBe(2);
    });
  });
});
