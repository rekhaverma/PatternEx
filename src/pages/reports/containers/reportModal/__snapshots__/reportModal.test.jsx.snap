// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportModal /> page:  render:  should match with snapshot 1`] = `
<Modal
  className="ptrx-modal"
  style={Object {}}
  type=""
>
  <section
    className="newReport"
  >
    <div
      className="newReport__row +spaceBetween newReport__firstrow"
    >
      <span
        className="newReport__title"
      >
        <FormattedMessage
          id="reports.reportModal.title"
          values={
            Object {
              "modalTitle": "",
            }
          }
        />
      </span>
      <span
        className="icon-close"
        onClick={[Function]}
      />
    </div>
    <div
      className="newReport__row +spaceBetween undefined"
    >
      <Input
        className="materialInput"
        errorMessage=""
        inputOptions={
          Object {
            "id": "reportName",
            "onChange": [Function],
            "type": "text",
            "value": undefined,
          }
        }
        label="*Name"
      />
      <div
        className="newReport__mode"
      >
        <SelectBox
          onClick={[Function]}
          options={
            Array [
              Object {
                "content": "Batch",
                "id": "batch",
              },
              Object {
                "content": "Realtime",
                "id": "realtime",
              },
            ]
          }
          placeholder="Mode"
          showLabel={true}
          singleSelect={true}
        />
      </div>
    </div>
    <div
      className="newReport__row +spaceBetween"
    >
      <div
        className="newReport__pipeline"
      >
        <SelectBox
          onClick={[Function]}
          options={Array []}
          placeholder="Select Pipeline"
          showLabel={true}
          singleSelect={true}
        />
      </div>
    </div>
    <div
      className="newReport__row +spaceBetween undefined"
    >
      <div
        className="newReport__rules"
      >
        <MultiSelect
          handleSelectChange={[Function]}
          onFooterClick={[Function]}
          options={Array []}
          placeholder="*Select Rules"
          type="customReportSpecific"
          value=""
        />
      </div>
    </div>
    <div
      className="newReport__row +rightSide"
    >
      <Button
        className="button--success +small"
        disabled={true}
        onClick={[Function]}
      >
        <FormattedMessage
          id="reports.save"
          values={Object {}}
        />
      </Button>
      <Button
        className="button--dark +small"
        onClick={[Function]}
      >
        <FormattedMessage
          id="reports.cancel"
          values={Object {}}
        />
      </Button>
    </div>
  </section>
</Modal>
`;
